File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        vector<int> dp(nums.size());
        dp[0]=nums[0];
        for(int i=1; i<nums.size(); i++){
            if(dp[i-1]>i-1){
                dp[i] = max(dp[i-1], nums[i]+i);
            }
            else return false;
        }
        return true;
    }
};

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int stepAcc = 1;
        for(int i=0; i<nums.length; i++){
            stepAcc--;
            if(stepAcc < 0) break;
            stepAcc = Math.max(nums[i], stepAcc);
            if(i==nums.length-1) return true;
        }
        return false;
    }
}